cas-server-core-tickets/src/main/java/org/jasig/cas/ticket/registry/AbstractDistributedTicketRegistry.java

```diff
@@ -38,11 +40,19 @@ protected final Ticket getProxiedTicketInstance(final Ticket ticket) {
             return null;
         }
 
+        if (ticket instanceof ProxyGrantingTicket) {
+            return new ProxyGrantingTicketDelegator(this, (ProxyGrantingTicket) ticket, needsCallback());
+        }
+
         if (ticket instanceof TicketGrantingTicket) {
-            return new TicketGrantingTicketDelegator(this, (TicketGrantingTicket) ticket, needsCallback());
+            return new TicketGrantingTicketDelegator<>(this, (TicketGrantingTicket) ticket, needsCallback());
+        }
+
+        if (ticket instanceof ProxyTicket) {
+            return new ProxyTicketDelegator(this, (ProxyTicket) ticket, needsCallback());
         }
 
-        return new ServiceTicketDelegator(this, (ServiceTicket) ticket, needsCallback());
+        return new ServiceTicketDelegator<>(this, (ServiceTicket) ticket, needsCallback());
```

Comment:
While this works, I'd rather you check also for ServiceTicket instances, and then throw errors if there is any other ticket types.